### PR TITLE
i18n(pt-BR): Add `directives-reference.md` translation

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [x] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

PT-BR translation for `directives-reference.md`

### CODE REVIEW

File: src/pages/pt-BR/reference/directives-reference.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+Por padrão, Astro irá processar, otimizar e fazer o bundle de tags `<script>` e `<style>` que ele ver na página. Você pode remover esse comportamento com a diretiva `is:inline`.
+
+`is:inline` diz ao Astro para deixar a tag `<script>` ou `<style>` como se fosse o HTML final resultante. Seus conteúdos não serão processados, otimizados ou passarão por bundle. Isso limita algumas funcionalidades do Astro, como importar pacotes npm ou utilizar uma linguagem que compila para CSS como Sass. 
+
+A diretiva `is:inline` siugnifica que tags `<style>` ou `<script>`:
+- Não passarão por bundle em um arquivo externo.
+- Não serão desduplicadas - o elemento irá aparecer quantas vezes ele for renderizado.
+- Não terá suas referências de `import`/`@import`/`url()` resolvidas relativamente ao arquivo `.astro`.
+- Não serão pré-processadas, por exemplo o atributo `<style lang="sass">` vai continuar gerando CSS puro.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:170]:
```suggestion
- Serão pré-processadas, por exemplo o atributo `<style lang="sass">` vai continuar gerando CSS puro.
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+- Não serão renderizadas no HTML final resultante exatamente como foram escritas.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:171]:
```suggestion
- Serão renderizadas no HTML final resultante exatamente como foram escritas.
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+- Estilos serão globais e não escopados ao componente.
+
+> ⚠️ A diretiva `is:inline` é implícita aonde qualquer outro atributo que não seja `src` é utilizado em uma tag `<script>` ou `<style>`.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:174]:
```suggestion
> ⚠️ A diretiva `is:inline` é implícita sempre que qualquer outro atributo que não seja `src` é utilizado em uma tag `<script>` ou `<style>`.
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  /* inline: importações relativas e de pacotes npm não são suportadas. */
+  console.log('Eu sou assim tanto aqui quanto no HTML final resultante.');
+</script>
+```
+
+📚 Veja como [scripts no lado do cliente](/pt-BR/core-concepts/astro-components/#scripts-no-lado-do-cliente) funcionam em componentes Astro.
+
+### `define:vars`
+
+`define:vars={...}` pode passar variáveis do frontmatter do seu componente no lado do servidor para o `<script>` ou `<style>` do cliente. Qualquer variável de frontmatter *serializável* é suportada, incluindo props passadas ao seu componente através de `Astro.props`.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:193]:
```suggestion
`define:vars={...}` pode passar variáveis do front matter do seu componente no lado do servidor para o `<script>` ou `<style>` do cliente. Qualquer variável de front matter *serializável* é suportada, incluindo props passadas ao seu componente através de `Astro.props`.
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+Utilizada internamente no componente `<Markdown />`.
+
+Por exemplo, se você tivesse um componente Katex customizado que converte algum texto para HTML, usuários poderiam fazer isso:
+
+```astro
+---
+import Katex from '../components/Katex.astro';
+---
+<Katex is:raw>SAlguma {sintaxe} conflitante aqui</Katex>
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:228]:
```suggestion
<Katex is:raw>Alguma {sintaxe} conflitante aqui</Katex>
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+Diretivas de template são utilizadas para controlar o comportamento de um elemento ou componente de alguma forma. Uma diretiva de template pode habilitar alguma funcionalidade do compilador que facilitaria sua vida (como utilizar `class:list` ao invés de `class`). Ou então, uma diretiva pode dizer para o compilador do Astro fazer algo especial com aquele componente (como hidratá-lo com `client:load`).
+
+Esta página descreve todas as diretivas de template disponíveis para você no Astro e como elas funcionam.
+
+## Regras
+
+Para uma diretiva de template ser válida, ela deve:
+
+- Incluir um dois-pontos `:` em seu nome, utilizando o padrão `X:Y` (ex: `client:load`).
+- Ser visível ao compilador (ex `X {...atributo}` não funcionaria se `atributo` contivesse uma diretiva).
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:18]:
```suggestion
- Ser visível ao compilador (ex.:`X {...atributo}` não funcionaria se `atributo` contivesse uma diretiva).
```

File: src/pages/pt-BR/reference/directives-reference.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+## Diretivas de Cliente
+
+Estas diretivas controlam como [componentes de Frameworks de UI](/pt-BR/core-concepts/framework-components/) são hidratados na página.
+
+Por padrão, um componente de framework de UI não é hidratado no cliente, Se nenhuma diretiva `client:*` é providenciada, seu HTML é renderizado na página sem nenhum JavaScript.
+
+### `client:load`
+
+- **Prioridade:** Alta
+- **Útil para:** Elementos de UI imediatamente visiveis que precisam ser interativos o mais rápido possível.
```
Comment (victor0x16) [inline src/pages/pt-BR/reference/directives-reference.md:86]:
```suggestion
- **Útil para:** Elementos de UI imediatamente visíveis que precisam ser interativos o mais rápido possível.
```

**Review by victor0x16 — CHANGES_REQUESTED** [review]

**Comment by victor0x16** [comment]
@Yan-Thomas, I told you :)

File: src/pages/pt-BR/reference/directives-reference.md [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+
+### `is:global`
+
+Por padrão, Astro automaticamente escopa as regras de CSS de `<style>` ao componente. Você pode remover esse comportamento com a diretiva `is:global`.
+
+`is:global` faz com que o conteúdo de uma tag `<style>` seja aplicado globalmente na página quando o componente está incluso. Isso desabilita o sistema de escopo de CSS do Astro. É o equivalente a envolver todos os seus seletores numa tag `<style>` com `:global()`.
+
+Você pode combinar `<style>` e `<style is:global>` no mesmo componente, para criar algumas regras de estilo globais enquanto ainda escopa a maioria do CSS do seu componente.
+
+Veja a página [Estilização & CSS](/pt-BR/guides/styling/#global-styles) para mais detalhas em como estilos globais funcionam.
```
[the remark this answers is not in the document]
Comment (Yan-Thomas) [reply src/pages/pt-BR/reference/directives-reference.md:152]:
```suggestion
Veja a página [Estilização & CSS](/pt-BR/guides/styling/#estilos-globais) para mais detalhas em como estilos globais funcionam.
```

**Comment by Yan-Thomas** [comment]
Thanks for the review @victor0x16!

I think I have killed the link checker somehow... :fearful:

**Comment by sarah11918** [comment]
Paging @hippotastic !

File: src/pages/pt-BR/core-concepts/astro-components.md [hunk 9]
```diff
@@ -371,7 +371,7 @@ Múltiplas tags `<script>` podem ser usadas no mesmo arquivo `.astro` combinando
 
 > **Nota:** Adicionar `type="module"` ou qualquer outro atributo em uma tag `<script>` irá desabilitar o comportamente padrão de empacotamento do Astro, tratando a tag como se houvesse a diretiva `is:inline`.
 
-📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference/#script--style-directives) para mais informação sobre as diretivas disponíveis em tags `<script>`.
+📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference#diretivas-de-script-e-estilização) para mais informação sobre as diretivas disponíveis em tags `<script>`.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/core-concepts/astro-components.md:374]:
```suggestion
📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference/#diretivas-de-script-e-estilização) para mais informação sobre as diretivas disponíveis em tags `<script>`.
```

**Comment by sarah11918** [comment]
So good news, I just updated the branch of another existing PR and the link checker completed just fine. So, it must be localized to here. I'll try re-running the test here. I've never seen it just fail this early before!  😅

File: src/pages/pt-BR/core-concepts/framework-components.md [hunk 10]
```diff
@@ -88,7 +88,7 @@ import ContagemInterativa from '../components/ContagemInterativa.jsx';
 
 Há diversas diretivas de hidratação disponíveis para componentes de frameworks de UI: `client:load`, `client:idle`, `client:visible`, `client:media={QUERY}` e `client:only={FRAMEWORK}`.
 
-📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference/#client-directives) para uma descrição completa destas diretivas de hidratação e seus usos.
+📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference#diretivas-de-cliente) para uma descrição completa destas diretivas de hidratação e seus usos.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/core-concepts/framework-components.md:91]:
```suggestion
📚 Veja nossa página de [referência de diretivas](/pt-BR/reference/directives-reference/#diretivas-de-cliente) para uma descrição completa destas diretivas de hidratação e seus usos.   
```

**Review by victor0x16 — APPROVED** [review]
LGTM

**Comment by hippotastic** [comment]
I don't see an issue. Logs look normal. The build has been cancelled by another build being started by the CI workflow. I'll probably separate the build step from the actual link check sometime to make it more clear that these are usually build issues and not link check issues. :)

**Pull request merged**